src/main/java/org/hyperledger/fabric/sdk/ServiceDiscovery.java

```diff
@@ -1280,17 +1281,20 @@ public String toString() {
 
     private static final int SERVICE_DISCOVER_FREQ_SECONDS = config.getServiceDiscoveryFreqSeconds();
 
+    private transient ScheduledExecutorService serviceDiscoveryExecutorService;
```

Comment:
I think this would be better (safer) as an AtomicReference rather than a transient field:
```java
private final AtomicReference<ScheduledExecutorService> serviceDiscoveryExecutorService = new AtomicReference<>();